movie-app/src/containers/app/App.tsx

```diff
@@ -1,60 +1,126 @@
-import HeaderContainer from "../header-container/HeaderContainer";
-import ListHeaderContainer from "../list-header-container/ListHeaderContainer";
-import MoviesListWrapper from "../movies-list-container/MoviesListWrapper";
-import MovieItem from "../../shared-components/movie-item/MovieItem";
-import MovieDetailsContainer from "../movie-details-container/MovieDetailsContainer";
-import SearchContainer from "../search-container/SearchContainer";
-import EmptyState from "../../shared-components/empty-state/EmptyState";
-import { IMovie } from "../../interfaces/IMovie";
-
-const MOVIE: IMovie = {
-  id: 0,
-  title: "Titanic",
-  tagline: null,
-  vote_average: 4,
-  vote_count: 0,
-  release_date: "1997",
-  poster_path: "../../assets/images/netflix-streaming-vs-traditional-cable.jpg",
-  overview:
-    "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
-  budget: 0,
-  revenue: 0,
-  runtime: 150,
-  genres: ["Action"]
-};
+import React, { useCallback, useEffect } from 'react';
+import { connect } from 'react-redux';
+import {
+  BrowserRouter as Router,
+  Redirect,
+  Route,
+  Switch,
+  Link,
+} from 'react-router-dom';
+import { IMovie } from '../../interfaces/IMovie';
+import {
+  getMovieById,
+  loadMovies,
+  setSearchParams,
+} from '../../redux/rootActions';
+import EmptyState from '../../shared-components/empty-state/EmptyState';
+import MovieItem from '../../shared-components/movie-item/MovieItem';
+import ErrorBoundary from '../error-boundary/ErrorBoundary';
+import HeaderContainer from '../header-container/HeaderContainer';
+import ListHeaderContainer from '../list-header-container/ListHeaderContainer';
+import MovieDetailsContainer from '../movie-details-container/MovieDetailsContainer';
+import MoviesListWrapper from '../movies-list-container/MoviesListWrapper';
+import SearchContainer from '../search-container/SearchContainer';
+import AppWrapper from './AppWrapper';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useQuery } from '../../constants/constants';
 
-const App = () => {
-  const [moviesList, getMoviesList] = useState();
+const App = (props: any) => {
+  const history = useHistory();
+  const location = useLocation();
+  const query = useQuery(location.search);
 
-  const getMovies = () => {
-    const arr: IMovie[] = [];
-    for (let index = 0; index <= 10; index++) {
-      arr.push(MOVIE);
+  useEffect(() => {
+    if (!history.location.pathname.includes('film')) {
```

Comment:
why is this taken from history? why not from location?

Reply:
done